### ISSUE TITLE
TypeError: rawRequest.connection.address is not a function

### ISSUE THREAD

**Issue by moltar** [Netflix/pollyjs#263]
## Description

We are using a library that uses a [Popsicle](https://github.com/serviejs/popsicle) client.

There is a failure in mocking the requests using Popsicle.

### Shareable Source

Pretty vanilla Polly setup:

```ts
import path from 'path'
import { Polly, PollyServer } from '@pollyjs/core'
import NodeHttpAdapter from '@pollyjs/adapter-node-http'
import FSPersister from '@pollyjs/persister-fs'

Polly.register(FSPersister)
Polly.register(NodeHttpAdapter)

export class PollyJS {
  recordingName: string
  polly: Polly

  constructor(recordingName: string) {
    this.recordingName = recordingName
    this.polly = new Polly(this.recordingName, {
      adapters: ['node-http'],
      persister: 'fs',
      persisterOptions: {
        fs: {
          recordingsDir: path.resolve(__dirname, '../__recordings__'),
        },
      },
    })
  }

  getPollyInstance(): Polly {
    return this.polly
  }

  getPollyServer(): PollyServer {
    return this.polly.server
  }
}
```

### Error Message & Stack Trace

```
    TypeError: rawRequest.connection.address is not a function

      at OverriddenClientRequest.onResponse (node_modules/popsicle-transport-http/src/index.ts:316:33)
      at _respond (node_modules/nock/lib/request_overrider.js:525:17)
      at respond (node_modules/nock/lib/request_overrider.js:508:11)
```

The code can be found here: https://github.com/serviejs/popsicle-transport-http/blob/ef78022e431532ec710c716ca91b7d3691afe888/src/index.ts#L316

### Config

See above.

### Dependencies

Copy the @pollyjs dependencies from `package.json`:

```json
{
    "@pollyjs/adapter-node-http": "2.6.3",
    "@pollyjs/core": "2.6.3",
    "@pollyjs/persister-fs": "2.6.3"
}
```

## Relevant Links

Provided by @nguyentoanit:
https://github.com/nguyentoanit/popsicle-polly-error

## Environment

Tell us which operating system you are using, as well as which versions of Node.js and npm/yarn. If applicable, include the browser and the corresponding version.

Run the following to get it quickly:

```
Node.js v10.15.3
darwin 18.7.0
npm 6.11.3
```

**Comment by nguyentoanit** [comment]
Minimal Project for reproduce error:
https://github.com/nguyentoanit/popsicle-polly-error

**Comment by offirgolan** [comment]
Thank you for the detailed issue @moltar and for the reproduction @nguyentoanit but this an issue with `nock` which we use under the hood. See nock/nock#1643.

**Comment by moltar** [comment]
@offirgolan thanks! Any suggested workarounds?

**Comment by offirgolan** [comment]
@moltar the best thing to do would be to open a PR to nock. We can also patch it on our end in the mean time but #264 needs to be addressed first.

**Comment by mastermatt** [comment]
This was fixed in Nock as of 11.7.0
